base/server/src/main/java/org/dogtagpki/server/cli/SubsystemRangeGeneratorUpdateCLI.java

```diff
@@ -0,0 +1,291 @@
+                if (nextBeginSerial != null && !nextBeginSerial.equals("-1")) {
+                    dbConfig.setNextBeginSerialNumber("0x" + nextBeginSerial);
+       
+                    LDAPEntry entryNextSerial = conn.read("cn=" + nextBeginSerial + "," + rangeDN);
+                    LDAPAttribute attrNextEnd = entryNextSerial.getAttribute("endRange");
+                    if (attrNextEnd != null) {
+                        nextEndSerial = attrNextEnd.getStringValues().nextElement();
+                    }
+                    dbConfig.setNextEndSerialNumber("0x" + nextEndSerial);
+                }
+
+                updateRanges(dbConfig, conn, baseDN, rangeDN, endSerialNumber, hostName, securePort);
+            } finally {
+                conn.disconnect();
+            }
+            return;
+        }
+        throw new EBaseException("Update to " + newGenerator + " not supported");
+    }
+
+    protected void updateRequestNumberRangeGenerator(PKISocketFactory socketFactory, LdapConnInfo connInfo,
+            LdapAuthInfo authInfo, DatabaseConfig dbConfig, String baseDN, IDGenerator newGenerator) throws EBaseException {
+        
+        String value = dbConfig.getString(
+                RequestRepository.PROP_REQUEST_ID_GENERATOR,
+                RequestRepository.DEFAULT_REQUEST_ID_GENERATOR);
+        idGenerator = IDGenerator.fromString(value);
+
+        if (newGenerator == IDGenerator.RANDOM && idGenerator != IDGenerator.RANDOM) {
+            logger.debug("Remove request ranges from configuration");
+            dbConfig.remove(DatabaseConfig.MIN_REQUEST_NUMBER);
+            dbConfig.remove(DatabaseConfig.MAX_REQUEST_NUMBER);
+            dbConfig.remove(DatabaseConfig.REQUEST_INCREMENT);
+            dbConfig.remove(DatabaseConfig.REQUEST_LOW_WATER_MARK);
+            dbConfig.remove(DatabaseConfig.REQUEST_CLONE_TRANSFER_NUMBER);
+            dbConfig.remove(DatabaseConfig.REQUEST_RANGE_DN);
+            dbConfig.put(RequestRepository.PROP_REQUEST_ID_GENERATOR, newGenerator.toString());
+            dbConfig.put(RequestRepository.PROP_REQUEST_ID_LENGTH, "128");
+            return;
+        }
+        if (newGenerator == IDGenerator.LEGACY_2 && idGenerator == IDGenerator.LEGACY) {
+            dbConfig.put(RequestRepository.PROP_REQUEST_ID_GENERATOR, newGenerator.toString());
+            return;
+        }
+        throw new EBaseException("Update to " + newGenerator + " not supported");
+    }
+    
+    private void updateRanges(DatabaseConfig dbConfig, LdapBoundConnection conn, String baseDN, String rangeDN, String defaultEndSerialNumber,
+        String hostName, String securePort) throws Exception{
+
+        LDAPSearchResults ranges = conn.search(rangeDN, LDAPv3.SCOPE_SUB, "(objectClass=pkiRange)", null, false);
+
+        BigInteger lastUsedSerial = BigInteger.ZERO;
+        boolean nextRangeToUpdate = true;
+        while (ranges.hasMoreElements()) {
```

Comment:
Could you add a note describing what this loop is doing? Is it trying to find the biggest `endRange` for the server?